consensus/mempool_test.go

```diff
@@ -134,28 +134,24 @@ func TestRmBadTx(t *testing.T) {
 			if r.GetCheckTx().Code != abci.CodeType_BadNonce {
 				t.Fatalf("expected checktx to return bad nonce, got %v", r)
 			}
-			cbCh <- struct{}{}
+			checkTxRespCh <- struct{}{}
 		})
 		if err != nil {
 			t.Fatal("Error after CheckTx: %v", err)
 		}
 
 		// check for the tx
-		for {
-			time.Sleep(time.Second)
-			txs := cs.mempool.Reap(1)
-			if len(txs) == 0 {
-				ch <- struct{}{}
-				return
-			}
-
+		txs := cs.mempool.Reap(1)
```

Comment:
how can we remove the for loop? what if the tx hasn't been removed yet

Reply:
would be best if we show the mempool is not empty first, and then it becomes empty when the abci app returns.

Reply:
looks like mempool is always empty